File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-4.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        maxF = 0
        count = {}
        l = 0
        res = 0

        for r in range(len(s)):
            count[s[r]] = 1 + count.get(s[r], 0)
            maxF = max(maxF, count[s[r]])

            if (r - l + 1) - maxF > k:
                count[s[l]] -= 1
                l += 1
            res = max(res, r - l + 1)

        return res

File: Data Structures & Algorithms/anagram-groups/submission-4.cs
public class Solution {
    public List<List<string>> GroupAnagrams(string[] strs) {
        var res = new Dictionary<string, List<string>>();
        foreach(var s in strs) {
            var count = new int[26];
            var charArray = s.ToCharArray();
            foreach(var c in charArray) {
                count[c - 'a']++;
                // c = a which is 97 => a - a = 97 - 97 = 0
                //count[0] = 1
            }
            var resultKey = string.Join(',', count);
            if(!res.ContainsKey(resultKey)) {
                res[resultKey] = new List<string>();
            }
            res[resultKey].Add(s);
        }      
        return res.Values.ToList<List<string>>();
    }
}